bbq/vm/vm.go

```diff
@@ -1008,6 +1008,12 @@ func loadTypeArguments(vm *VM, typeArgs []uint16) []bbq.StaticType {
 func maybeDereference(context interpreter.ValueStaticTypeContext, value Value) Value {
 	switch typedValue := value.(type) {
 	case *interpreter.EphemeralReferenceValue:
+		// Do not dereference attachments, so that the receiver is a reference as expected.
+		if val, ok := typedValue.Value.(*interpreter.CompositeValue); ok {
+			if val.Kind == common.CompositeKindAttachment {
+				return value
+			}
+		}
```

Comment:
Given this is specific to references to receivers, let's maybe rename this function to `maybeDereferenceReceiver`, so it is not accidentally used for other purposes unrelated to receivers